apitools/base/py/credentials_lib.py

```diff
@@ -48,65 +48,57 @@ def SetCredentialsCacheFileLock(lock):
     cache_file_lock = lock
 
 
-# TODO(craigcitro): Expose the extra args here somewhere higher up,
-# possibly as flags in the generated CLI.
+# List of additional methods we use when attempting to construct
+# credentials. Users can register their own methods here, which we try
+# before the defaults.
+_CREDENTIALS_METHODS = []
```

Comment:
that's interesting to hear -- i didn't realize boto also did things this way, but it makes sense.

my primary need is to invert the dependencies in some internal code; i could actually make these methods internal-only for now (they're not listed in `__all__` yet, on purpose), if you think i'm going to regret it ...